src/GogeToken.sol

```diff
@@ -596,346 +584,350 @@ contract DogeGaySon is ERC20, Ownable {
+
+        emit TeamWalletUpdated(teamWallet, _newWallet);
     }
     
     function updateMarketingWallet(address _newWallet) external {
-        require(_msgSender() == DAO || _msgSender() == owner(), "Not authorized");
-        require(_newWallet != marketingWallet, "The marketing wallet is already this address");
+        require(_msgSender() == gogeDao || _msgSender() == owner(), "GogeToken.sol::updateMarketingWallet() msg.sender is not owner or gogeDao");
+        require(_newWallet != marketingWallet, "GogeToken.sol::updateMarketingWallet() address is already set");
+
         isExcludedFromFees[_newWallet] = true;
-        emit MarketingWalletUpdated(marketingWallet, _newWallet);
         marketingWallet = _newWallet;
+
+        emit MarketingWalletUpdated(marketingWallet, _newWallet);
     }
     
-    function setSwapTokensAtAmount(uint256 _swapAmount) external {
-        require(_msgSender() == DAO || _msgSender() == owner(), "Not authorized");
+    function updateSwapTokensAtAmount(uint256 _swapAmount) external {
+        require(_msgSender() == gogeDao || _msgSender() == owner(), "GogeToken.sol::updateSwapTokensAtAmount() msg.sender is not owner or gogeDao");
         swapTokensAtAmount = _swapAmount * (10**18);
     }
     
-    function setTradingIsEnabled(uint256 botBlocks, uint8 botFees) external onlyOwner {
-        require(tradingIsEnabled == false, "Trading is already enabled");
+    function enableTrading() external onlyOwner {
+        require(tradingIsEnabled == false, "GogeToken.sol::enableTrading() trading is already enabled");
+
         cakeDividendRewardsFee = 10;
         marketingFee = 2;
-        buyBackAndLiquidityFee = 2;
+        buyBackFee = 2;
         teamFee = 2;
         totalFees = 16;
         marketingEnabled = true;
-        buyBackAndLiquifyEnabled = true;
+        buyBackEnabled = true;
         cakeDividendEnabled = true;
         teamEnabled = true;
-        swapTokensAtAmount = 20000000 * (10**18);
+        swapTokensAtAmount = 20_000_000 * (10**18);
         tradingIsEnabled = true;
-        _botBlocks = botBlocks;
-        _botFees = botFees;
         _firstBlock = block.timestamp;
+
+        emit TradingEnabled(_firstBlock);
     }
     
-    function setBuyBackAndLiquifyEnabled(bool _enabled) external {
-        require(_msgSender() == DAO || _msgSender() == owner(), "Not authorized");
-        require(buyBackAndLiquifyEnabled != _enabled, "Can't set flag to same status");
-        if (_enabled == false) {
-            previousBuyBackAndLiquidityFee = buyBackAndLiquidityFee;
-            buyBackAndLiquidityFee = 0;
-            buyBackAndLiquifyEnabled = _enabled;
+    function setBuyBackEnabled(bool _enabled) external {
+        require(_msgSender() == gogeDao || _msgSender() == owner(), "GogeToken.sol::setBuyBackEnabled() not authorized");
+        require(buyBackEnabled != _enabled, "GogeToken.sol::setBuyBackEnabled() can't set flag to same status");
+
+        if (!_enabled) {
```

Comment:
We should not be checking negated booleans, this is unnecessary. Change this for all the fee assignments and anything else doing this. Should save gas.
```suggestion
        if (_enabled) {
            buyBackFee = previousbuyBackFee;
            buyBackEnabled = _enabled;
        } else {
            previousbuyBackFee = buyBackFee;
            buyBackFee = 0;
            buyBackEnabled = _enabled;
        }
```